contributing.md

```diff
@@ -0,0 +1,275 @@
+```
+
+A *fork* is created using the web UI. It may then be cloned for a user
+called 'username' as follows,
+
+``` sh
+git clone git@github.com:username/janus-core.git janus-core-username
+```
+
+### Branch, fix, merge model:
+
+All work should follow the workflow of branch, fix, merge. Let us assume
+you have an issue with your code which needs to be fixed.
+
+#### Step 1: Branch from your fork
+
+Create a new branch for the issue on the dashboard of your fork, we will
+assume the branch is called 'issueXYZ'. Clone the branch,
+
+``` sh
+$ git clone -b issueXYZ --single-branch git@github.com:username/janus-core.git janus-issueXYZ
+```
+
+Alternatively you can create the branch in the cli using
+
+``` sh
+# clone the repository, if you already have a local repository this is not nessecary
+$ git clone git@github.com:username/janus-core.git janus-issueXYZ
+$ pushd janus-issueXYZ
+# create and checkout a new branch (this is equivilent to git branch followed by git checkout)
+$ git checkout -b issueXYZ
+# create a remote tracking branch for you to push your changes to
+$ git push -u origin issueXYZ
+```
+
+#### Step 2: Fix the issue and commit your changes
+
+Fix whatever is wrong. Use git status to see which files you have
+changed and prepare a commit.
+
+``` sh
+# stage changes
+$ git add <filename|folder> to add the new things
+# commit the changes with a clear and brief message
+$ git commit -m "<commit message>"
+# push the commit to origin
+$ git push
+```
+
+#### Step 3a: Merge your branch into upstream
+
+Follow the provided link in after the push and continue the web interface
+Add any relevant labels or milestones and
+assign a reviewer. Compare the code and if you are happy click to submit your pull request.
+
+After the pull request has been submitted tests will be run and your
+reviewer will be notified.
+
+#### Step 3b: Finalising the merge
+
+If all is OK with the commit your reviewer may set the request to be
+merged once all tests pass. Otherwise the reviewer may open discussions
+using the Github comment system to point out issues that may need to be
+addressed before the commit can be merged.
+
+If changes need to be made you may make more commits onto your branch.
+When you push your branch to your *fork* the pull request will be
+automatically updated to use the latest commit. Reply to the discussions
+to indicate when and how they have been addressed.
+
+If your branch has become out of sync with *upstream* then conflicts may
+arise. Sometimes these cannot be automatically resolved and you will
+need to resolve them by hand. Github provides instructions for this, or
+you can follow this routine,
+
+``` sh
+# add upstream as a remote if you have not already
+$ git remote add upstream git@github.com:stfc/janus-core.git
+# get the changes to upstream since you started working on your issue
+$ git fetch upstream
+# merge these changes into your branch (assuming you want to merge into the master branch on devel)
```

Comment:
```suggestion
# merge these changes into your branch (assuming you want to merge into the main branch on devel)
```
Also not sure `devel` has been introduced at this point? I can also imagine people might think it's a branch name, since it's not uncommon.